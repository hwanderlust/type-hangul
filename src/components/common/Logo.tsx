import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

import { StyledProps } from "../../helpers";
import logo from "../../images/logo.png";

const Button = styled(Link)`
  display: block;
`;
const Icon = styled.img`
  height: 100px;
  transform: translateY(-10%);

  @media only screen and (max-width: 720px) {
    height: 50px;
    transform: translateY(-25%);
  }
`;

function Logo(props: StyledProps) {
  return (
    <Button {...props} to="/">
      <Icon src={logo} alt="a 'T' with the Korean ã…Ž as a logo" />
    </Button>
  );
}

export default Logo;