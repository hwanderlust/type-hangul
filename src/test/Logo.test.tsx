import 'jest-styled-components';
import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Logo from "../components/common/Logo";

describe("<Logo/>", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Logo />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

