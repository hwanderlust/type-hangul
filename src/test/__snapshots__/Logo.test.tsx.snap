// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> renders correctly 1`] = `
.c0 {
  display: block;
}

.c1 {
  height: 100px;
  -webkit-transform: translateY(-10%);
  -ms-transform: translateY(-10%);
  transform: translateY(-10%);
}

@media only screen and (max-width:720px) {
  .c1 {
    height: 50px;
    -webkit-transform: translateY(-25%);
    -ms-transform: translateY(-25%);
    transform: translateY(-25%);
  }
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
>
  <img
    alt="a 'T' with the Korean ã…Ž as a logo"
    className="c1"
    src="logo.png"
  />
</a>
`;
